tests/lib/utils.js

```diff
@@ -110,11 +125,35 @@ describe('utils', () => {
           meta: null,
           isNewStyle: false,
         },
+
+        // ESM (object style)
+        'export default { create() {} }': {
+          create: { type: 'FunctionExpression' },
+          meta: null,
+          isNewStyle: true,
+        },
+        'export default { create() {}, meta: {} }': {
+          create: { type: 'FunctionExpression' },
+          meta: { type: 'ObjectExpression' },
+          isNewStyle: true,
+        },
+
+        // ESM (function style)
+        'export default function () {}': {
+          create: { type: 'FunctionDeclaration' },
+          meta: null,
+          isNewStyle: false,
+        },
+        'export default () => {}': {
+          create: { type: 'ArrowFunctionExpression' },
+          meta: null,
+          isNewStyle: false,
+        },
       };
 
       Object.keys(CASES).forEach(ruleSource => {
         it(ruleSource, () => {
-          const ast = espree.parse(ruleSource, { ecmaVersion: 6, range: true });
+          const ast = espree.parse(ruleSource, { ecmaVersion: 6, range: true, sourceType: ruleSource.includes('export default') ? 'module' : 'script' });
```

Comment:
it seems very not reliable to just check 'export default'.  imagine this comment: `// export default xxxx`.

can we get parser options from eslint here?

Reply:
I tweaked it a bit to check that the test case starts with `export default` which should be safer. I know it's not great, but this is just for a single test.